src/planner/binder/statement/bind_create.cpp

```diff
@@ -177,6 +178,89 @@ static void FindMatchingPrimaryKeyColumns(vector<unique_ptr<Constraint>> &constr
 	throw BinderException("there is no primary key for referenced table \"%s\"", fk.info.table);
 }
 
+void ExpressionContainsGeneratedColumn(const ParsedExpression &expr, const unordered_set<string> &gcols,
+                                       bool &contains_gcol) {
+	if (contains_gcol) {
+		return;
+	}
+	if (expr.type == ExpressionType::COLUMN_REF) {
+		auto &column_ref = (ColumnRefExpression &)expr;
+		auto &name = column_ref.GetColumnName();
+		if (gcols.count(name)) {
+			contains_gcol = true;
+			return;
+		}
+	}
+	ParsedExpressionIterator::EnumerateChildren(
+	    expr, [&](const ParsedExpression &child) { ExpressionContainsGeneratedColumn(child, gcols, contains_gcol); });
+}
+
+static bool AnyConstraintReferencesGeneratedColumn(CreateTableInfo &table_info) {
+	unordered_set<string> generated_columns;
+	for (auto &col : table_info.columns) {
+		if (!col.Generated()) {
+			continue;
+		}
+		generated_columns.insert(col.Name());
+	}
+	if (generated_columns.empty()) {
+		return false;
+	}
+
+	for (auto &constr : table_info.constraints) {
+		switch (constr->type) {
+		case ConstraintType::CHECK: {
+			auto &constraint = (CheckConstraint &)*constr;
+			auto &expr = constraint.expression;
+			bool contains_generated_column = false;
+			ExpressionContainsGeneratedColumn(*expr, generated_columns, contains_generated_column);
+			if (contains_generated_column) {
+				return true;
+			}
+			break;
+		}
+		case ConstraintType::FOREIGN_KEY: {
+			auto &constraint = (ForeignKeyConstraint &)*constr;
+			// Can't/dont have to check foreign keys?
+			for (auto &key : constraint.pk_columns) {
+				if (generated_columns.count(key)) {
+					return true;
```

Comment:
This line is never run

Reply:
Exception would have been thrown earlier if this was the case, so I've removed this check